utils.py

```diff
@@ -51,26 +51,52 @@ def get_scenario_params_list(config):
     """
 
     scenario_params_list = []
+    # Separate scenarios from different dataset
+    config_dataset = []
 
     for list_scenario in config:
+        
+        if isinstance(list_scenario['dataset_name'], dict):
+            for dataset_name in list_scenario['dataset_name'].keys():
+                # Add path to init model from an existing model
+                dataset_scenario = list_scenario.copy()
+                dataset_scenario['dataset_name'] = [dataset_name]
+                if list_scenario['dataset_name'][dataset_name] is None:
+                    dataset_scenario['init_model_from'] = ['random_initialization']
+                else:
+                    dataset_scenario['init_model_from'] = list_scenario['dataset_name'][dataset_name]
+                
+                config_dataset.append(dataset_scenario)
+        else:
+            config_dataset.append(list_scenario)
+    
+    for list_scenario in config_dataset:
         params_name = list_scenario.keys()
         params_list = list(list_scenario.values())
 
         for el in product(*params_list):
             scenario = dict(zip(params_name, el))
-
+        
             if scenario['partners_count'] != len(scenario['amounts_per_partner']):
                 raise Exception("Length of amounts_per_partner does not match number of partners.")
 
             if scenario['samples_split_option'][0] == 'advanced' \
                     and (scenario['partners_count'] != len(scenario['samples_split_option'][1])):
                 raise Exception("Length of samples_split_option does not match number of partners.")
-
+        
             if 'corrupted_datasets' in params_name:
                 if scenario['partners_count'] != len(scenario['corrupted_datasets']):
                     raise Exception("Length of corrupted_datasets does not match number of partners.")
-
+            
+            # keep only init_model_from if a path is given.
+            try:
+                if scenario['init_model_from'] == 'random_initialization':
+                    del scenario['init_model_from']
+            except KeyError:
+                pass
```

Comment:
Ok... question: why not create it in the first place? (L65)

Reply:
The block implemented L65 write the list of parameters so that the Cartesian product can be applied cleanly.